medicai/utils/cam.py

```
@@ -0,0 +1,252 @@
+        value_and_grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+
+        # Compute hidden activations and gradients
+        hidden_output = hidden_model(input_tensor)
+        (_, predictions), grads = value_and_grad_fn(hidden_output)
+        return grads, hidden_output, predictions
+
+    def _compute_gradients_torch(self, input_tensor, target_class_index, mask_type):
+        try:
+            import torch
+        except ImportError:
+            raise ImportError("PyTorch is not installed.")
```

Ruff (0.14.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)